### PR TITLE
Refactor email token decoding

### PR DESCRIPTION
**Pull request by matthew-robinson-ons**
# What and why?
When put_email_verification or verify_token encounter an exception during the decoding of the email token, it is raised within the session and causes incorrect logging and session rollbacks. 

This PR moves the decoding of the email token outside of the session context for both functions and reduced duplicate code within the functions. 

Flask-cors also had a vulnerability which has been updated on this pr. 
# How to test?
Ensure email verification still works correctly when updating or registering an email and the calls to the party service no longer contain session rollbacks or incorrect logging when an exception is thrown. 
# Jira
https://jira.ons.gov.uk/secure/RapidBoard.jspa?rapidView=1493&projectKey=RAS&view=detail&selectedIssue=RAS-1262

### CODE REVIEW

File: ras_party/controllers/account_controller.py [hunk 1]
```diff
@@ -389,18 +389,34 @@ def _send_account_email_changed_notification(email_address, new_email_address, r
...
+@with_query_only_db_session
+def verify_token(token, session):
+    """
+    Verify the email token and check if that respondent exists.
+    :param token: The email token for verification
+    :param session: Database session
+    :return: Verification response
+    """
 
+    email_address = decode_token(token)
```
Comment (LJBabbage) [inline ras_party/controllers/account_controller.py]: You're still inside the decorator and have just refactored the function, put_email_verification should just be getting an email, the token decode should be done outside session scope

Comment (matthew-robinson-ons) [reply ras_party/controllers/account_controller.py]: Have moved the decode outside of the function now. Finding it difficult to test this change but believe the recent changes have solved it.

**Review by anwilkie — APPROVED** [review]
Looks good to me after testing.

**Review by LJBabbage — APPROVED** [review]

**Pull request merged**